tests/PersonDialogTest.java

```diff
@@ -0,0 +1,204 @@
+import org.junit.jupiter.api.Test;
+import org.junit.rules.TemporaryFolder;
+
+import static java.awt.event.KeyEvent.*;
+
+
+import java.io.IOException;
+
+class PersonDialogTest {
+
+    @Rule
+    private static FrameFixture window = null;
+    public static TemporaryFolder folder = new TemporaryFolder();
+
+
+    @BeforeAll
+    public static void init() {
+        // Prevent program exiting
+        NoExitSecurityManagerInstaller.installNoExitSecurityManager();
+
+        // Required for full AssertJ GUI testing
+        FailOnThreadViolationRepaintManager.install();
+    }
+
+    @BeforeEach
+    void setUp() throws IOException, ClassNotFoundException{
+        // Initialize window
+        AddressBookGUI frame = GuiActionRunner.execute(() -> new AddressBookGUI());
+        window = new FrameFixture(frame);
+        window.show();
+    }
+
+    @AfterEach
+    void tearDown() {
+        // Close assertJ window gui
+        window.cleanUp();
+    }
+
+    @AfterAll
+    public static void clean() {
+        // Re-enable program to close after testing completes
+        NoExitSecurityManagerInstaller.installNoExitSecurityManager().uninstall();
+    }
+   
+    /**
+     * This tests that a person can be added.
+     */
+
+    @Test
+    void addPersonValid() {
+        window.button("add").click();
+        DialogFixture dialog = window.dialog();
+
+        // Type 'Muffin','Man','1234 Dreary Lane','Cookie City','WA','66666', and '1234567890'
+        // into the respective boxes
+        dialog.textBox("firstName").pressKey(VK_SHIFT).pressAndReleaseKeys(VK_M).releaseKey(VK_SHIFT)
+                .pressAndReleaseKeys(VK_U, VK_F, VK_F, VK_I, VK_N);
+        dialog.textBox("lastName").pressKey(VK_SHIFT).pressAndReleaseKeys(VK_M).releaseKey(VK_SHIFT)
+                .pressAndReleaseKeys(VK_A, VK_N);
+        dialog.textBox("address").pressAndReleaseKeys(VK_1, VK_2, VK_3, VK_4, VK_SPACE).pressKey(VK_SHIFT)
+                .pressAndReleaseKeys(VK_D).releaseKey(VK_SHIFT)
+                .pressAndReleaseKeys(VK_R, VK_E, VK_A, VK_R, VK_Y, VK_SPACE).pressKey(VK_SHIFT)
+                .pressAndReleaseKeys(VK_L).releaseKey(VK_SHIFT).pressAndReleaseKeys(VK_A, VK_N, VK_E);
+        dialog.textBox("city").pressKey(VK_SHIFT).pressAndReleaseKeys(VK_C).releaseKey(VK_SHIFT)
+                .pressAndReleaseKeys(VK_O, VK_O, VK_K, VK_I, VK_E, VK_SPACE).pressKey(VK_SHIFT).pressAndReleaseKeys(VK_C)
+                .releaseKey(VK_SHIFT).pressAndReleaseKeys(VK_I, VK_T, VK_Y);
+        dialog.textBox("state").pressKey(VK_SHIFT).pressAndReleaseKeys(VK_W, VK_A).releaseKey(VK_SHIFT);
+        dialog.textBox("zip").pressAndReleaseKeys(VK_6, VK_6, VK_6, VK_6, VK_6);
+        dialog.textBox("phone").pressAndReleaseKeys(VK_1, VK_2, VK_3, VK_4, VK_5, VK_6, VK_7, VK_8, VK_9, VK_0);
+
+        // Click 'OK'
+        dialog.button(JButtonMatcher.withText("OK")).click();
+
+        // Test person is added
+        window.table().requireRowCount(1);
+
+    }
```

Comment:
This looks like it is just a copy of my test. You should be skipping the part about the AddressBookGUI and just testing the PersonDialog. This seems like a great candidate for the mock/stub. Try loading the PersonDialog directly in the @beforeeach function